code/obj/submachine/cooking.dm

```diff
@@ -146,6 +161,56 @@ TYPEINFO(/obj/submachine/chef_sink)
 		..()
 
 
+/datum/action/bar/private/handwashing/critterwashing
+	duration = 7 DECI SECONDS
+	id = "critterwashing"
+	var/mob/living/critter/small_animal/victim
+	var/obj/item/grab/grab
+	var/datum/aiTask/timed/wandering
+	New(usermob,sink,critter,thegrab)
+		boutput(world, critter)
+		victim = critter
+		grab = thegrab
+		..()
+
+	checkStillValid()
+		if(GET_DIST(victim, sink) > 0 || BOUNDS_DIST(user, sink) > 1 || victim == null || user == null || sink == null || !grab)
+			interrupt(INTERRUPT_ALWAYS)
+			return FALSE
+		return TRUE
+	onStart()
+		if (istype(victim, /mob/living/critter/small_animal/cat) && victim.ai?.enabled)
+			victim._ai_patience_count = 0
```

Comment:
The leading underscore feels like this is meant to be a private variable to me. Is it really ok to modify it like this?

Reply:
it's not great but there's really no other way i could spot, i could make a setter function if that's a better looking outcome